apps/s_cb.c

```diff
@@ -759,7 +759,7 @@ int verify_cookie_callback(SSL *ssl, const unsigned char *cookie,
 int generate_stateless_cookie_callback(SSL *ssl, unsigned char *cookie,
                                        size_t *cookie_len)
 {
-    unsigned int temp;
+    unsigned int temp = 0;
     int res = generate_cookie_callback(ssl, cookie, &temp);
     *cookie_len = temp;
```

Comment:
Formally speaking this is not correct solution. I mean original code was arguably formally incorrect, in sense that it would be more appropriate to not touch *cookie_len in case error is returned. In other words more appropriate way to handle this as
```
int res = generate_cookie_callback(ssl, cookie, &temp);
if (res)
    *cookie_len = temp;
```

Reply:
Though it's not given that it would actually resolve the warning. This is because generate_cookie_callback is not static function and if code is compiled with -fPIC, compiler would have to assume that it can be overridden at link time. This effectively means that it would have to "consider" generate_stateless_cookie_callback by itself, without "looking" at generate_cookie_callback. Bottom line is that one might still have to initialize temp.